src/resource/solidDataset.ts

```diff
@@ -1135,3 +1138,41 @@ function resolveLocalIrisInThing(
       : thing.url,
   });
 }
+
+/**
+ * Fetch the contents of '.well-known/solid' for a given resource URL.
+ *
+ * The contents of the '.well-known/solid' endpoint define the capabilities of the server, and provide their associated endpoints/locations.
+ * This behaves similarly to the use of '.well-known' endpoints in e.g. (OIDC servers)[https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig]
+ *
+ * @param url URL of a Resource.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns Promise resolving to a [[SolidDataset]] containing the data at '.well-known/solid' for the given Resource, or rejecting if fetching it failed.
+ */
+export async function getWellKnownSolid(
+  url: UrlString | Url,
+  options: Partial<
+    typeof internal_defaultFetchOptions & ParseOptions
+  > = internal_defaultFetchOptions
+): Promise<SolidDataset & WithServerResourceInfo> {
+  const urlString = internal_toIriString(url);
+  const resourceMetadata = await getResourceInfo(urlString, options);
+  const linkedResources = getLinkedResourceUrlAll(resourceMetadata);
+  const rootResources = linkedResources[pim.storage];
+  const rootResource = rootResources?.length === 1 ? rootResources[0] : null;
+  if (!rootResource) {
```

Comment:
```suggestion
  if (rootResource === null) {
```
isn't testing for the value we expect more specific that testing for nullish ?